documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -1777,3 +1778,28 @@ Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rul
 **How to fix the violation**: Provide detailed description of the node in the description section.
 
 Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [ARM](#arm-violations): [Errors](#arm-errors) or [Warnings](#arm-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R4001" ></a>R4001 XmsParameterLocation
+
+**Category** : SDK Error
+
+**Applies to** : ARM and Data Plane OpenAPI(swagger) specs
+
+**Output Message** : The parameter 'managementGroupId' is defined in global parameters section without 'x-ms-parameter-location' extension. This would add the parameter as the client property. Please ensure that this is exactly you want. If so, apply the extension "x-ms-parameter-location": "client". Else, apply the extension "x-ms-parameter-location": "method".
```

Comment:
Seems like 'managementGroupId' is a specific value, was this intended? I'm guessing you meant for this to be a placeholder.

Reply:
/as a client property/

Reply:
How do I know what I want? What is the alternative if I don't want it added as a client property? That's not quite clear to me. Maybe something like:
"If you're sure you want to expose this parameter as a property on the client itself, apply the extension... If you don't want the parameter to be exposed on the client and intend it to be passed only to relevant method(s), instead apply the extension ..."

At least I think that's what's the two options are, I'm still not super clear even after reading the below.

Reply:
> Seems like 'managementGroupId' is a specific value, was this intended? I'm guessing you meant for this to be a placeholder.

Good catch, fixed locally :)

> How do I know what I want? What is the alternative if I don't want it added as a client property? That's not quite clear to me.

This is good feedback. In regards to changing the output text, we'll have to go create a follow-up issue in the tooling repository. However, I do want to ensure that after reading this documentation that the choice is clear. I'll rephrase to drive clarity.